File: internal/store/analytics.go
package store

import "golang-asses/internal/models"

type AnalyticsStore struct {
	RevenueByCountry map[string]float64
	ProductCount     map[string]int
	RegionRevenue    map[string]float64
	MonthlySales     [12]int
}

func NewAnalyticsStore() *AnalyticsStore {
	return &AnalyticsStore{
		RevenueByCountry: make(map[string]float64),
		ProductCount:     make(map[string]int),
		RegionRevenue:    make(map[string]float64),
	}
}

func (s *AnalyticsStore) Add(txn models.Transaction) {
	s.RevenueByCountry[txn.Country] += txn.Revenue
	s.ProductCount[txn.Product] += txn.Quantity
	s.RegionRevenue[txn.Region] += txn.Revenue
	s.MonthlySales[txn.Date.Month()-1]++

}

File: web/src/types/analytics.ts
// API Response Types matching Go backend structures

export type RevenueByCountry = Record<string, number>;

export type ProductCount = Record<string, number>;

export type RegionRevenue = Record<string, number>;

export type MonthlySales = number[];

// Derived types for UI display
export interface CountryRevenue {
  country: string;
  revenue: number;
}

export interface ProductSales {
  product: string;
  quantity: number;
}

export interface RegionData {
  region: string;
  revenue: number;
}

export interface MonthlySalesData {
  month: string;
  sales: number;
}

// Dashboard summary stats
export interface DashboardStats {
  totalCountries: number;
  totalRegions: number;
  totalProducts: number;
  totalTransactions: number;
}

// API state for each resource
export interface AsyncState<T> {
  data: T | null;
  isLoading: boolean;
  error: string | null;
}

